README.md

```diff
@@ -0,0 +1,159 @@
+
+For background on device-coap-c, CoAP and low power wireless, see the [presentation](https://zoom.us/rec/share/N2Uh7C9qScsj32bs0T8aNF4VPPOuFSypnhQp3g2LmSFfOA16giRq9gwqpGvNb1HX.kknLyNV7Rj72mPms?startTime=1602514686000) to the Device Working Group.
+
+## Resources
+
+device-coap creates a parameterized CoAP resource to which data may be posted:
+
+```
+   /a1r/{deviceName}/{resourceName}
+```
+
+- `a1r` is short for "API v1 resource", as defined by device-rest-go.
+- `deviceName` refers to a `device` managed by the CoAP device service. For example, `res/configuration.toml` pre-defines a device named 'd1'.
+- `resourceName` refers to a `deviceResource` defined in the device profile, as described in the sub-section below.
+
+Payload data posted to one of these resources is type validated, and the resulting value then is sent into EdgeX via the Device SDK's asynchronous `post_readings` capability.
+
+## Profiles
+
+[example-datatype.yaml](./res/example-datatype.yaml) defines  generic resources for data types. The table below shows the available resource names and correspondence with CoAP attributes. 
+
+For example, the 'int' resource name means that EdgeX provides a CoAP resource, `/a1r/{deviceName}/int`. This resource accepts an integer encoded as text, like `42`.
+
+| resourceName | Type   | EdgeX MediaType<br>CoAP Content-Format|
+|---------|--------|---------------------------------------|
+| int     | Int32  | text/plain                            |
+| float   | Float64| text/plain                            |
+| json    | String | application/json                      |
+
+>_Note:_ You must define the Content-Format option in the CoAP POST request. See the _Testing_ section below for example use.
+
+
+## Configuration
+
+This section describes properties in [configuration.toml](./res/configuration.toml) as used by device-coap. See the _Configuration and Registry_ section of the EdgeX documentation for background.
+
+### Driver
+
+Below are the recognized properties for the Driver section, followed by an example.
+
+| Key         | Value                                                                             |
+|-------------|-----------------------------------------------------------------------------------|
+| CoapBindAddr| Address on which CoAP server listens for devices                                  |
+| SecurityMode| DTLS client-server security type. Does not support raw public key or certificates.|
+| PskKey      | Pre-shared key. Accepts only a single key, ignored in NoSec mode.                 |
+
+
+```
+[Driver]
+  # Supports IPv4 or IPv6 if provided by network infrastructure. Use '0.0.0.0'
+  # for any IPv4 interface, or '::' for any IPv6 interface.
+  CoapBindAddr = '0.0.0.0'
+  # Choose 'PSK' or 'NoSec'
+  SecurityMode = 'PSK'
+  # Key is up to 16 arbitrary bytes; must be base64 encoded here
+  PskKey = 'ME42aURHZ3Uva0Y0eG9lZw=='
+```
+
+### DeviceList
+The `DeviceList` section pre-defines the 'd1' device.
```

Comment:
Call out the fact that no ProtocolProperties are define for CoAP devices (perhaps there will be in future if writes are implenented?)

Reply:
Yes, I expect we may need to define some ProtocolProperties for a device, particularly for addressing writes.

Fixed in 9fa2601